### ISSUE TITLE
Spravit zmeny v testovacich DB na zaklade navrhu zmeny perzistentnej vrstvy

### ISSUE THREAD

**Issue by Juraj326** [TIS2024-FMFI/gastro#60]
![Image](https://github.com/user-attachments/assets/d662bb86-4e53-4b04-9068-7c6ffd112dac)

**Comment by Juraj326** [comment]
Issues, ktore vyzaduju tuto zmenu:
* #41 
* #44 
* #45 
* #49 
* #51 
* #54 
* #58

**Comment by Juraj326** [comment]
- [ ] business_trips.spp_symbol_id_1
- [ ] business_trips.spp_symbol_id_2
- [ ] business_trips.spp_symbol_id_3
- [ ] business_trips.amount_eur_1
- [ ] business_trips.amount_eur_2
- [ ] business_trips.amount_eur_3
- [ ] business_trips.isTemplate
- [ ] spp_symbols.Agency
- [ ] spp_symbols.Acronym
- [x] countries.trips_count
- [ ] users.iban
- [ ] spp_user_type

**Comment by Juraj326** [comment]
# `countries.trips_count`

```mysql

ALTER TABLE `cesty_new`.`countries`
ADD COLUMN `tmp` bigint unsigned NOT NULL DEFAULT '0' AFTER `trips_count`;

```

**Comment by Juraj326** [comment]
# `countries.trips_count`

```mysql

ALTER TABLE `cesty_new`.`countries`
ADD COLUMN `trips_count` bigint unsigned NOT NULL DEFAULT '0' AFTER `name`;

```

**Comment by Juraj326** [comment]
Bolo by dobre pridat do DB pritomnosti do tabulky `absence` column `cesty_business_trip_sofia_id` na identifikaciu pracovnych ciest. Momentalne sa identifikuju dvojicou `user_id`, `date_time`, co nie je uplne optimalne z dovodu, ze v pracovnych cestach je to implementovane inak a integracia je kvoli tomu potom narocnejsia. Pridanie identifikatora podla sofia_id by sa to vyrazne zlahcilo.

**Comment by Juraj326** [comment]
Navrhujem implementovat column `cesty_business_trip_sofia_id` nasledovne:

```mysql
ALTER TABLE `pritomnost_new`.`absence`
ADD COLUMN `sofia_id` varchar(40) NOT NULL DEFAULT '0000' AFTER `confirmation`;
```

**Comment by Juraj326** [comment]
> Bolo by dobre pridat do DB pritomnosti do tabulky `absence` column `cesty_business_trip_sofia_id` na identifikaciu pracovnych ciest. Momentalne sa identifikuju dvojicou `user_id`, `date_time`, co nie je uplne optimalne z dovodu, ze v pracovnych cestach je to implementovane inak a integracia je kvoli tomu potom narocnejsia. Pridanie identifikatora `cesty_business_trip_sofia_id` by sa to vyrazne zlahcilo.

> Navrhujem implementovat column `cesty_business_trip_sofia_id` nasledovne:
> 
> ALTER TABLE `pritomnost_new`.`absence`
> ADD COLUMN `cesty_business_trip_sofia_id ` varchar(40) NOT NULL DEFAULT '-' AFTER `confirmation`;


Pridal som tento stlpec do nasej testovacej db:

```mysql
ALTER TABLE `pritomnost_new`.`absence`
ADD COLUMN `cesty_business_trip_sofia_id` varchar(40) NOT NULL DEFAULT '-' AFTER `confirmation`;
```

**Comment by Gubledon** [comment]
# `spp_symbols`


```mysql


ALTER TABLE `cesty_new`.`spp_symbols`
    ADD COLUMN `agency` VARCHAR(100) NOT NULL DEFAULT '' AFTER `status`,
    ADD COLUMN `acronym` VARCHAR(10) NOT NULL DEFAULT '' AFTER `agency`,
    DROP COLUMN `fund`;

```

**Comment by Gubledon** [comment]
# `business_trips.state`


```mysql

UPDATE `cesty_new`.`business_trips`
SET `state` = `state` - 1
WHERE `state` >= 2;

```

**Comment by Gubledon** [comment]
# `business_trips`


```mysql

ALTER TABLE business_trips
    ADD COLUMN spp_symbol_id_2 BIGINT UNSIGNED NULL AFTER reimbursement_id,
    ADD COLUMN spp_symbol_id_3 BIGINT UNSIGNED NULL AFTER spp_symbol_id_2,
    ADD COLUMN amount_eur SMALLINT NULL AFTER spp_symbol_id_3,
    ADD COLUMN amount_eur_2 SMALLINT NULL AFTER amount_eur,
    ADD COLUMN amount_eur_3 SMALLINT NULL AFTER amount_eur_2,
    ADD COLUMN is_template BOOLEAN NOT NULL DEFAULT FALSE AFTER conclusion;

ALTER TABLE business_trips
    ADD CONSTRAINT business_trips_spp_symbol_id_2_foreign 
        FOREIGN KEY (spp_symbol_id_2) REFERENCES spp_symbols(id),
    ADD CONSTRAINT business_trips_spp_symbol_id_3_foreign 
        FOREIGN KEY (spp_symbol_id_3) REFERENCES spp_symbols(id);

```

**Comment by Gubledon** [comment]
# `users`


```mysql

ALTER TABLE `cesty_new`.`users`
ADD COLUMN iban VARCHAR(34) NULL AFTER remember_token,
ADD COLUMN spp_user_type SMALLINT UNSIGNED NULL;


```

**Comment by Cerulikova** [comment]
Tu je už asi všetko urobené, tak to zavriem.

**Comment by Juraj326** [comment]
```sql
ALTER TABLE `cesty_new`.`users`
ADD COLUMN `personal_id_dochadzka` int NULL AFTER `updated_at`;
```

**Comment by Juraj326** [comment]
> # `Users`
> 
> ALTER TABLE `cesty_new`.`users`
> ADD COLUMN `personal_id_dochadzka` int NULL AFTER `updated_at`;

asi by bolo dobre upravit ten moqups db

**Comment by Cerulikova** [comment]
moqups som upravila

**Comment by Cerulikova** [comment]
> # `spp_symbols`
> 
> ALTER TABLE `cesty_new`.`spp_symbols`
>     ADD COLUMN `agency` VARCHAR(100) NOT NULL AFTER `status`,
>     ADD COLUMN `acronym` VARCHAR(10) NOT NULL AFTER `agency`,
>     DROP COLUMN `fund`;

Kedže chceme, aby polia agency a acronym mohli byť aj prázdne, musíme povoliť NULL

**Comment by LeaMichalikova** [comment]
> > # `spp_symbols`
> > ALTER TABLE `cesty_new`.`spp_symbols`
> > ADD COLUMN `agency` VARCHAR(100) NOT NULL AFTER `status`,
> > ADD COLUMN `acronym` VARCHAR(10) NOT NULL AFTER `agency`,
> > DROP COLUMN `fund`;
> 
> Kedže chceme, aby polia agency a acronym mohli byť aj prázdne, musíme povoliť NULL. Upravila som testovaciu DB.

Vporiadku, predpokladám že stači dať preč "NOT NULL" z oboch riadkov?

**Comment by Gubledon** [comment]
Ano, zmeny tu v komentoch editnem a do issue [#67](https://github.com/TIS2024-FMFI/gastro/issues/67) pridám opravený alter script.

**Comment by Gubledon** [comment]
Hotovo, alter script bol upravený, zatváram issue
